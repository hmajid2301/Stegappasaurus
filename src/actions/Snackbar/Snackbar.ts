import Snackbar from 'react-native-snackbar';

import {primary, pureWhite} from '~/constants/colors';

interface Show {
  text: string;
  buttonText?: string;
  onButtonPress?: () => void;
  duration?: number;
}

export default class CustomSnackbar {
  public static show({
    onButtonPress = () => null,
    buttonText = 'Okay',
    duration = 5000,
    text,
  }: Show) {
    Snackbar.show({
      action: {
        color: primary,
        onPress: onButtonPress,
        title: buttonText,
      },
      color: pureWhite,
      duration,
      title: text,
    });
  }
}
