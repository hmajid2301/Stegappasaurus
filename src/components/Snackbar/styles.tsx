import { StyleSheet } from "react-native";

import { colors, fonts } from "~/common/styles";

const styles = StyleSheet.create({
  buttonText: { color: colors.primary },
  snackbar: { backgroundColor: "#323232", borderRadius: 3, margin: 5 },
  text: { fontSize: 14, fontFamily: fonts.bodyLight, paddingHorizontal: 10 }
});

export default styles;
