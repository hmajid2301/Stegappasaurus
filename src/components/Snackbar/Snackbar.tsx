import Snackbar from "react-native-snackbar";

import { colors } from "~/constants";

interface IShow {
  text: string;
  buttonText?: string;
  onButtonPress?: () => void;
  duration?: number;
}

export default class CustomSnackbar {
  public static show = ({
    onButtonPress = () => null,
    buttonText = "Okay",
    duration = 5000,
    text
  }: IShow) => {
    Snackbar.show({
      action: {
        color: colors.primary,
        onPress: onButtonPress,
        title: buttonText
      },
      duration,
      title: text
    });
  };
}
