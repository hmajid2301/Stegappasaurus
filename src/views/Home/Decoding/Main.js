import { ImagePicker, Permissions } from 'expo';
import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';

import styles from './Main/styles';


class Decoding extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
    togglePrimaryColor: PropTypes.func.isRequired,
  };

  componentDidMount = () => {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  }

  getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  selectPhotoToEncode = (uri) => {
    this.props.navigation.navigate('DecodeImage', { uri });
  }

  render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <TouchableOpacity onPress={this.getPhotoFromCameraRoll} style={styles.button}>
          <Icon name='photo' style={styles.icon} type='FontAwesome' />
        </TouchableOpacity>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Decoding);
