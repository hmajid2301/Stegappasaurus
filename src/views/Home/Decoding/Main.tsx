import * as ImagePicker from "expo-image-picker";
import * as Permissions from "expo-permissions";
import * as React from "react";
import { TouchableOpacity, View } from "react-native";
import { Icon } from "react-native-elements";
import { NavigationScreenProp } from "react-navigation";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import { ITheme, PrimaryColorNames } from "@types";
import PhotoAlbumList from "~/components/PhotoAlbumList";
import Snackbar from "~/components/Snackbar";
import { PRIMARY_COLORS } from "~/constants";
import { togglePrimaryColor } from "~/redux/actions";

import styles from "./Main/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
  togglePrimaryColor: (primaryColor: PrimaryColorNames) => void;
}

class Main extends React.Component<IProps, {}> {
  public render() {
    const { theme } = this.props.screenProps;
    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity
            onPress={this.getPhotoFromCameraRoll}
            style={styles.button}
          >
            <Icon name="photo" iconStyle={styles.icon} type="font-awesome" />
          </TouchableOpacity>
        </View>

        <View style={styles.photoListContainer}>
          <PhotoAlbumList onPhotoPress={this.selectPhotoToEncode} />
        </View>
      </View>
    );
  }

  public componentDidMount = () => {
    this.props.navigation.addListener("willFocus", () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.BLUE.name);
    });
  };

  private getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === "granted") {
      const result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.Images
      });
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      Snackbar.show({
        text: "This app does not have permission to access the camera roll."
      });
    }
  };

  private selectPhotoToEncode = (uri: string) => {
    this.props.navigation.navigate("Progress", { uri });
  };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  togglePrimaryColor: (colorName: PrimaryColorNames) =>
    dispatch(togglePrimaryColor({ color: colorName }))
});

export default connect(
  null,
  mapDispatchToProps
)(Main);
