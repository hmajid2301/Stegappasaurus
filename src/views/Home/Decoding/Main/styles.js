import { Dimensions, StyleSheet } from 'react-native';

import { colors } from '~/util/styles';


const pageHeight = Dimensions.get('window').height;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },

  button: {
    backgroundColor: colors.secondary,
    height: pageHeight / 2,
    justifyContent: 'center',
    margin: 5,
    borderRadius: 10,
  },

  icon: {
    color: colors.pureWhite,
    fontSize: 150,
    textAlign: 'center',
  },

});

export default styles;
