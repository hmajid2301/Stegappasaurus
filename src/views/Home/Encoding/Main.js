import { ImagePicker, MediaLibrary, Permissions } from 'expo';
import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  ActivityIndicator,
  FlatList,
  Image,
  TouchableOpacity,
  View,
} from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';
import { colors } from '~/util/styles';

import styles from './Main/styles';


class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      photos: [],
      loading: false,
      lastPhoto: 0,
    };
  }

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
    togglePrimaryColor: PropTypes.func.isRequired,
  }

  componentDidMount = async () => {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.ORANGE.name);
    });

    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
        first: 15,
        sortBy: [MediaLibrary.SortBy.creationTime],
      });
      this.setState({ photos: assets, lastPhoto: endCursor });
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  padData = (data) => {
    const numOfCols = 3;
    const fullRows = Math.floor(data.length / numOfCols);
    let numElementsInFinalRow = data.length - (fullRows * numOfCols);

    while (numElementsInFinalRow !== numOfCols && numElementsInFinalRow !== 0) {
      data.push({ empty: true });
      numElementsInFinalRow += 1;
    }

    return data;
  };

  getPhotoFromCamera = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA, Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchCameraAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera permission not granted');
    }
  }

  getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  getPhotoFromCatAPI = () => {
    this.setState({ loading: true });

    const url = 'https://api.thecatapi.com/v1/images/search?mime_types=jpg,png';
    fetch(url)
      .then(response => response.json())
      .then((responseJson) => {
        this.selectPhotoToEncode(responseJson[0].url);
      })
      .catch();

    setTimeout(() => {
      this.setState({ loading: false });
    }, 3000);
  }

  getMorePhotosFromCameraRoll = async () => {
    const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
      first: 9,
      sortBy: [MediaLibrary.SortBy.creationTime],
      after: this.state.lastPhoto,
    });

    this.setState({
      photos: [...this.state.photos, ...assets],
      photoEnd: endCursor,
    });
  }

  renderPhotosFromCameraRoll = ({ item }) => {
    if (item.empty === true) {
      return <View/>;
    }
    return (
      <TouchableOpacity
        onPress={() => this.selectPhotoToEncode(item.uri)}
        style={styles.photoButton}
      >
        <Image source={{ uri: item.uri }} style={styles.photos}/>
      </TouchableOpacity>
    );
  }

  selectPhotoToEncode = (uri) => {
    this.props.navigation.navigate('EncodeImage', { uri });
  }

  render() {
    const { theme } = this.props.screenProps;

    if (this.state.loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator color={colors.primary} size='large'/>
        </View>
      );
    }

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity onPress={this.getPhotoFromCamera} style={styles.button}>
            <Icon name='camera' style={styles.icon} type='FontAwesome' />
          </TouchableOpacity>
          <TouchableOpacity onPress={this.getPhotoFromCameraRoll} style={styles.button}>
            <Icon name='photo' style={styles.icon} type='FontAwesome'/>
          </TouchableOpacity>
          <TouchableOpacity onPress={this.getPhotoFromCatAPI} style={styles.button}>
            <Icon name='cat' style={styles.icon} type='MaterialCommunityIcons'/>
          </TouchableOpacity>
        </View>
        <View style={styles.photoListContainer}>
          <FlatList
            data={this.padData(this.state.photos)}
            keyExtractor={(_, index) => index}
            numColumns={3}
            onEndReached={this.getMorePhotosFromCameraRoll}
            renderItem={this.renderPhotosFromCameraRoll}
          />
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Main);
