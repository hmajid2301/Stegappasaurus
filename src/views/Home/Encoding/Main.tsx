import { create } from "apisauce";
import { ImagePicker, MediaLibrary, Permissions } from "expo";
import { Icon } from "native-base";
import React, { Component } from "react";
import {
  ActivityIndicator,
  Alert,
  FlatList,
  Image,
  TouchableOpacity,
  View
} from "react-native";
import env from "react-native-dotenv";
import { NavigationScreenProp } from "react-navigation";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import { PRIMARY_COLORS } from "~/common/constants";
import { ITheme, PrimaryColorNames } from "~/common/interfaces";
import { colors } from "~/common/styles";
import { togglePrimaryColor } from "~/redux/actions";

import styles from "./Main/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
  togglePrimaryColor: (primaryColor: PrimaryColorNames) => void;
}

interface IState {
  photos: IPhoto[];
  loading: boolean;
  lastPhoto: number;
}

interface IPhoto {
  uri: string;
}

interface ICatAPI {
  breeds: string[];
  id: string;
  url: string;
  width: number;
  height: number;
}

class Main extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      lastPhoto: 0,
      loading: false,
      photos: []
    };
  }

  public render() {
    const { theme } = this.props.screenProps;

    if (this.state.loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator color={colors.primary} size="large" />
        </View>
      );
    }

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity
            onPress={this.getPhotoFromCamera}
            style={styles.button}
          >
            <Icon name="camera" style={styles.icon} type="FontAwesome" />
          </TouchableOpacity>

          <TouchableOpacity
            onPress={this.getPhotoFromCameraRoll}
            style={styles.button}
          >
            <Icon name="photo" style={styles.icon} type="FontAwesome" />
          </TouchableOpacity>

          <TouchableOpacity
            onPress={this.getPhotoFromCatAPI}
            style={styles.button}
          >
            <Icon
              name="cat"
              style={styles.icon}
              type="MaterialCommunityIcons"
            />
          </TouchableOpacity>
        </View>
        <View style={styles.photoListContainer}>
          <FlatList
            data={this.padData(this.state.photos)}
            keyExtractor={(item, index) => item.uri + index}
            numColumns={3}
            onEndReached={this.morePhotosFromCameraRoll}
            renderItem={this.renderPhotosFromCameraRoll}
          />
        </View>
      </View>
    );
  }

  public componentDidMount = async () => {
    this.props.navigation.addListener("willFocus", () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.ORANGE.name);
    });

    const { status } = await Permissions.askAsync(
      Permissions.CAMERA,
      Permissions.CAMERA_ROLL
    );
    if (status === "granted") {
      const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
        first: 15,
        sortBy: [MediaLibrary.SortBy.creationTime]
      });
      this.setState({ photos: assets, lastPhoto: endCursor });
    } else {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera roll.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private padData = (data: IPhoto[]) => {
    const numOfCols = 3;
    const fullRows = Math.floor(data.length / numOfCols);
    let numElementsInFinalRow = data.length - fullRows * numOfCols;

    while (numElementsInFinalRow !== numOfCols && numElementsInFinalRow !== 0) {
      data.push({ uri: "" });
      numElementsInFinalRow += 1;
    }

    return data;
  };

  private getPhotoFromCamera = async () => {
    try {
      const result = await ImagePicker.launchCameraAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } catch {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private getPhotoFromCameraRoll = async () => {
    try {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } catch {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera roll.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private getPhotoFromCatAPI = async () => {
    this.setState({ loading: true });

    const api = create({
      baseURL: "https://api.thecatapi.com",
      headers: { "x-api-key": env.CAT_API_KEY }
    });

    const response = await api.get("/v1/images/search?mime_types=jpg,png");
    if (response.ok) {
      const urls = response.data as ICatAPI[];
      this.selectPhotoToEncode(urls[0].url);
    } else {
      Alert.alert(
        "Failed",
        "Failed to get a cat photo, check you're connected to the internet.",
        [
          {
            text: "ok"
          }
        ]
      );
    }

    setTimeout(() => {
      this.setState({ loading: false });
    }, 1000);
  };

  private morePhotosFromCameraRoll = async () => {
    const { assets } = await MediaLibrary.getAssetsAsync({
      after: this.state.lastPhoto,
      first: 9,
      sortBy: [MediaLibrary.SortBy.creationTime]
    });

    this.setState({
      photos: [...this.state.photos, ...assets]
    });
  };

  private renderPhotosFromCameraRoll = ({ item }: { item: IPhoto }) => {
    if (item.uri === "") {
      return <View />;
    }
    return (
      <TouchableOpacity
        onPress={() => this.selectPhotoToEncode(item.uri)}
        style={styles.photoButton}
      >
        <Image source={{ uri: item.uri }} style={styles.photos} />
      </TouchableOpacity>
    );
  };

  private selectPhotoToEncode = (uri: string) => {
    this.props.navigation.navigate("Message", { uri });
  };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  togglePrimaryColor: (colorName: PrimaryColorNames) =>
    dispatch(togglePrimaryColor({ color: colorName }))
});

export default connect(
  null,
  mapDispatchToProps
)(Main);
