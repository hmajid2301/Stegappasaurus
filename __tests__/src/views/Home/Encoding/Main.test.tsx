import {render, fireEvent, wait} from '@testing-library/react-native';
import React from 'react';
import ImagePicker from 'react-native-image-picker';

import Snackbar from '~/actions/Snackbar';
import Main from '~/views/Home/Encoding/Main';

const navigation: any = {
  addListener: jest.fn(),
  navigate: jest.fn(),
  getParam: jest.fn(),
  state: {
    routeName: 'Main',
  },
};

describe('Encoding Main: Functionality', () => {
  test('Open Camera Pass', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    ImagePicker.launchCamera = jest.fn().mockImplementation((_, callback) =>
      callback({
        uri: 'file:///storage/emulated/0/Stegappasaurus/1575927505003.png',
      }),
    );

    const spy = jest.spyOn(navigation, 'navigate');
    const touchable = getByTestId('camera');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith('Message', {
        uri: 'file:///storage/emulated/0/Stegappasaurus/1575927505003.png',
      });
    });
  });

  test('Open Camera Fail', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    ImagePicker.launchCamera = jest.fn().mockImplementation(() => {
      throw new Error();
    });

    const spy = jest.spyOn(Snackbar, 'show');
    const touchable = getByTestId('camera');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith({
        text: 'This app does not have permission to access the camera.',
      });
    });
  });

  test('Open Camera Roll Pass', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    ImagePicker.launchImageLibrary = jest
      .fn()
      .mockImplementation((_, callback) =>
        callback({
          uri: 'file:///storage/emulated/0/Stegappasaurus/1575927505003.png',
        }),
      );

    const spy = jest.spyOn(navigation, 'navigate');
    const touchable = getByTestId('cameraroll');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith('Message', {
        uri: 'file:///storage/emulated/0/Stegappasaurus/1575927505003.png',
      });
    });
  });

  test('Open Camera Roll Fail', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    ImagePicker.launchImageLibrary = jest.fn().mockImplementation(() => {
      throw new Error();
    });

    const spy = jest.spyOn(Snackbar, 'show');
    const touchable = getByTestId('cameraroll');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith({
        text: 'This app does not have permission to access the camera roll.',
      });
    });
  });

  test('Open thecatapi Pass', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    (fetch as jest.Mock).mockReturnValue({
      json: jest.fn().mockReturnValue({
        status: 200,
        data: [
          {
            breeds: 'bengal',
            id: '1',
            url: 'www.example.com/cat.png',
            width: 100,
            height: 100,
          },
        ],
      }),
    });

    const spy = jest.spyOn(navigation, 'navigate');
    const touchable = getByTestId('catapi');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith('Message', {
        uri: 'file:///storage/emulated/0/Stegappasaurus/1575927505003.png',
      });
    });
  });

  test('Open thecatapi Fail', async () => {
    const {getByTestId} = render(<Main navigation={navigation} />);

    (fetch as jest.Mock).mockReturnValue({
      json: jest.fn().mockReturnValue({
        status: 404,
      }),
    });

    const spy = jest.spyOn(Snackbar, 'show');
    const touchable = getByTestId('cameraroll');
    fireEvent.press(touchable);

    await wait(() => {
      expect(spy).toHaveBeenCalledWith({
        text:
          "Failed to get a cat photo, check you're connected to the internet.",
      });
    });
  });
});
