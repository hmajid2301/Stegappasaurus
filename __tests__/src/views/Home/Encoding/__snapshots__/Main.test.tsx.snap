// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encoding Main: Match Snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <Loader
    loading={false}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 2,
        "marginTop": 2,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="camera"
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="photo"
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="cat"
        type="material-community"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 2,
      }
    }
  >
    <PhotoAlbumList
      onPhotoPress={[Function]}
    />
  </View>
</View>
`;

exports[`Encoding Main: Match Snapshots 2 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#17212D",
      },
    ]
  }
>
  <Loader
    loading={false}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 2,
        "marginTop": 2,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="camera"
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="photo"
        type="font-awesome"
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#009CFF",
          "flex": 1,
          "height": 250,
          "justifyContent": "center",
          "marginLeft": 2,
          "marginTop": 2,
        }
      }
    >
      <Themed.Icon
        iconStyle={
          Object {
            "color": "#FFF",
            "fontSize": 40,
            "textAlign": "center",
          }
        }
        name="cat"
        type="material-community"
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 2,
      }
    }
  >
    <PhotoAlbumList
      onPhotoPress={[Function]}
    />
  </View>
</View>
`;

exports[`Encoding Main: Redux mapDispatchToProps 1`] = `
Array [
  Object {
    "payload": Object {
      "color": "ORANGE",
    },
    "type": "TOGGLE_PRIMARY_COLOR",
  },
]
`;
